orchestration/flows/bl832/job_controller.py

```diff
@@ -0,0 +1,287 @@
+from abc import ABC, abstractmethod
+from dotenv import load_dotenv
+import logging
+import os
+from pathlib import Path
+from typing import Optional
+
+from orchestration.flows.bl832.config import Config832
+from orchestration.nersc import NerscClient
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+load_dotenv()
+
+
+class TomographyHPCController(ABC):
+    """
+    Abstract class for tomography HPC controllers.
+    Provides interface methods for reconstruction and building multi-resolution datasets.
+
+    Args:
+        ABC: Abstract Base Class
+    """
+    def __init__(
+        self,
+        Config832: Optional[Config832] = None
+    ) -> None:
+        pass
+
+    @abstractmethod
+    def reconstruct(
+        self,
+        file_path: str = "",
+    ) -> bool:
+        """Perform tomography reconstruction
+
+        :param file_path: Path to the file to reconstruct.
+        :return: True if successful, False otherwise.
+        """
+        pass
+
+    @abstractmethod
+    def build_multi_resolution(
+        self,
+        file_path: str = "",
+    ) -> bool:
+        """Generate multi-resolution version of reconstructed tomography
+
+        :param file_path: Path to the file for which to build multi-resolution data.
+        :return: True if successful, False otherwise.
+        """
+        pass
+
+
+class ALCFTomographyHPCController(TomographyHPCController):
+    """
+    Implementation of TomographyHPCController for ALCF.
+    Methods here leverage Globus Compute for processing tasks.
+
+    Args:
+        TomographyHPCController (ABC): Abstract class for tomography HPC controllers.
+    """
+
+    def __init__(self) -> None:
+        pass
+
+    def reconstruct(
+        self,
+        file_path: str = "",
+    ) -> bool:
+
+        # uses Globus Compute to reconstruct the tomography
+        pass
+
+    def build_multi_resolution(
+        self,
+        file_path: str = "",
+    ) -> bool:
+        # uses Globus Compute to build multi-resolution tomography
+
+        pass
+
+
+class NERSCTomographyHPCController(TomographyHPCController):
```

Comment:
NERSC-specific code can now be moved into `/flows/bl832/nersc.py`

Reply:
I tested this more thoroughly after moving `NERSCTomographyHPCController` definition into `/flows/bl832/nersc.py`, and unfortunately this leads to a circular import issue.

The issue arises because `get_controller()`/`HPC(Enum)` relies on importing the specific controller implementations. Since the flow scripts contain the controller-specific implementation and require importing `get_controller()` to kick off the flows, the circular import ensues. I can see if there is a cleaner way to approach this, as I get why we would want to define those classes in their respective nersc.py and alcf.py files. Otherwise, I will fallback to keeping the classes defined in `job_controller.py` to avoid the error.